### DISCUSSION TITLE
Exceptions while trying to store secrets using environment variables option

### DISCUSSION THREAD

**Discussion by nboddu009** [Kong/kong#12942] [Help] [answered]
Hi Team, 
Can someone help with below issue,
Issue : could not get value from external vault (no value found)

**Description:**

`Installed Kong enterprise version with database mode (kong-enterprise-edition_3.6.1.1_amd64.deb)
Trying use to storing secrets in environment variables(https://docs.konghq.com/gateway/3.6.x/kong-enterprise/secrets-management/backends/env/)
But It is giving me error like 'could not get value from external vault (no value found)'

**Steps Produce the issue:**
	1. export SECRETS_PASSWORD=abc123
	2. export KONG_VAULT_ENV_PREFIX=SECRETS_
	3. Created Prefix using Admin API
	curl -i -X PUT http://hostname:8001/vaults/my-env-vault \
	 --data name=env \
	 --data description="Store secrets in environment variables" \
	--data config.prefix="SECRETS_"
   4. sudo kong reload
   5. sudo kong vault get my-env-vault/secrets-password
	
   Error: 
	User1:~$ export SECRETS_PASSWORD=abc123
	User1:~$ export KONG_VAULT_ENV_PREFIX=SECRETS_
	User1:~$ sudo kong reload
	2024/04/25 16:10:38 [warn] ulimit is currently set to "1024". For better performance set it to at least "4096" using "ulimit -n"
	2024/04/25 16:10:39 [warn] ulimit is currently set to "1024". For better performance set it to at least "4096" using "ulimit -n"
	Kong reloaded
	User1:~$ sudo kong vault get my-env-vault/secrets-password
	2024/04/25 16:11:49 [warn] ulimit is currently set to "1024". For better performance set it to at least "4096" using "ulimit -n"
	Error: could not get value from external vault (no value found)
	
	  Run with --v (verbose) or --vv (debug) for more details
	User1:~$ sudo kong vault get my-env-vault/password
	2024/04/25 16:11:59 [warn] ulimit is currently set to "1024". For better performance set it to at least "4096" using "ulimit -n"
	Error: could not get value from external vault (no value found)
	
	  Run with --v (verbose) or --vv (debug) for more details`

**Comment by StarlightIbuki** [comment, marked as answer]
It should be `my-env-vault/password` instead of `my-env-vault/secrets-password` given your configuration.
